staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -1537,3 +1585,62 @@ func (g *fixtureTestGen) Init(c *generator.Context, w io.Writer) error {
 	}
 	return nil
 }
+
+// extractInnerValidations extracts all +k8s:inner validations for the given
+// subfield that were defined on the parent struct.The syntax for the tag is
+// +k8s:inner(subfield-name)=<validator-tag>=<args>
+func (td *typeDiscoverer) extractInnerValidations(subfield *types.Member, comments []string) (validators.Validations, error) {
+	var result validators.Validations
+
+	// Currently the format for +k8s:inner tag is:
+	// +k8s:inner(subfield-name)=<validator-tag>=<args> validator tag args..."
```

Comment:
`+k8s:ifOptionEnabled` syntax is `+k8s:ifOptionEnabled=FeatureX=+k8s:validateFalse="field T1.S1"` which is kind of awful but works.  Aesthetically I prefer your notation (as long as we have one arg, and no real parsing).

Should we convert options to `+k8s:ifOptionEnabled(FeatureX)=+k8s:validateFalse="field T1.S1" ?

@jpbetz